src/test-project/src/test/java/com/redhat/devtools/intellij/commonUiTestLibrary/fixturesTest/dialogs/projectManipulation/NewProjectDialogTest.java

```diff
@@ -11,145 +11,389 @@
         assertTrue(currentlySelectedProjectSdk.contains("8"), "Selected project SDK should be Java 8 but is '" + currentlySelectedProjectSdk + "'");
-        newProjectDialog.setProjectSdkIfAvailable("11");
-        currentlySelectedProjectSdk = listOfRemoteTextToString(projectJdkComboBox.findAllText());
+        newProjectFirstPage.setProjectSdkIfAvailable("11");
+        currentlySelectedProjectSdk = TextUtils.listOfRemoteTextToString(projectJdkComboBox.findAllText());
         assertTrue(currentlySelectedProjectSdk.contains("11"), "Selected project SDK should be Java 11 but is '" + currentlySelectedProjectSdk + "'");
     }
 
     @Test
     public void selectNewProjectTypeTest() {
-        newProjectDialog.selectNewProjectType("Empty Project");
-        boolean isEmptyProjectLabelVisible = !newProjectDialog.findAll(JListFixture.class, byXpath("//div[@visible_text='Empty Project']")).isEmpty();
+        newProjectFirstPage.selectNewProjectType("Empty Project");
+        boolean isEmptyProjectLabelVisible = !newProjectFirstPage.findAll(JListFixture.class, byXpath("//div[@visible_text='Empty Project']")).isEmpty();
         assertTrue(isEmptyProjectLabelVisible, "The 'Empty Project' label should be visible but is not.");
 
-        newProjectDialog.selectNewProjectType("Java FX");
-        boolean isJavaFXApplicationLabelVisible = !newProjectDialog.findAll(JListFixture.class, byXpath("//div[@visible_text='JavaFX Application']")).isEmpty();
+        newProjectFirstPage.selectNewProjectType("Java FX");
+        boolean isJavaFXApplicationLabelVisible = !newProjectFirstPage.findAll(JListFixture.class, byXpath("//div[@visible_text='JavaFX Application']")).isEmpty();
         assertTrue(isJavaFXApplicationLabelVisible, "The 'Java FX' label should be visible but is not.");
     }
 
     private void navigateToSetProjectNamePage(NewProjectType newProjectType) {
-        newProjectDialog.selectNewProjectType(newProjectType.toString());
-        newProjectDialog.next();
+        newProjectFirstPage.selectNewProjectType(newProjectType.toString());
+        newProjectDialogWizard.next();
         if (newProjectType == NewProjectType.PLAIN_JAVA) {
-            newProjectDialog.next();
+            newProjectDialogWizard.next();
+        }
+    }
+
+    private void testProjectNameInputField(NewProjectType newProjectType) {
+        navigateToSetProjectNamePage(newProjectType);
+        AbstractNewProjectFinalPage finalPage;
+
+        switch (newProjectType) {
+            case PLAIN_JAVA:
+                finalPage = newProjectDialogWizard.find(JavaNewProjectFinalPage.class, Duration.ofSeconds(10));
+                break;
+            case MAVEN:
+            case GRADLE:
+                finalPage = newProjectDialogWizard.find(MavenGradleNewProjectFinalPage.class, Duration.ofSeconds(10));
+                break;
+            default:
+                throw new UITestException("Unsupported project type.");
         }
```

Comment:
I have seen this functionality being used on other place as well. It would really be nice to have this extracted and reused.

Reply:
Yes, I see it now. Fixed.